modules/drivers/mongo/test/metabase/driver/mongo/parameters_test.clj

```diff
@@ -235,4 +300,84 @@
+                                                             {:$sort {:_id 1}}
+                                                             {:$limit 10}])
+                                            :collection    "venues"
+                                            :template-tags {"price" {:name "price"
+                                                                     :display-name "Price"
+                                                                     :type         :dimension
+                                                                     :dimension    $price}}}
+                               :parameters [{:type   operator
+                                             :target [:dimension [:template-tag "price"]]
+                                             :value  input}]})))))))
+        (testing "variadic operators"
+          (let [run-query! (fn [operator]
+                             (mt/dataset geographical-tips
+                               (mt/rows
+                                 (qp/process-query
+                                   (mt/query tips
+                                     {:type       :native
+                                      :native     {:query         (json/generate-string
+                                                                   [{:$match (json-raw "{{username}}")}
+                                                                    {:$sort {:_id 1}}
+                                                                    {:$project {"username" "$source.username"}}
+                                                                    {:$limit 20}])
+                                                   :collection    "tips"
+                                                   :template-tags {"username" {:name         "username"
+                                                                               :display-name "Username"
+                                                                               :type         :dimension
+                                                                               :dimension    $tips.source.username}}}
+                                      :parameters [{:type   operator
+                                                    :target [:dimension [:template-tag "username"]]
+                                                    :value  ["bob" "tupac"]}]})))))]
+            (is (= #{"bob" "tupac"}
+                   (into #{} (map second)
+                         (run-query! :string/=))))
+            ;; the dis-equality aren't variadic yet
+            (is (= #{}
+                     (set/intersection
+                      #{"bob" "tupac"}
+                      ;; most of these are nil as most records don't have a username. not equal is a bit ambiguous in
+                      ;; mongo. maybe they might want present but not equal semantics
+                      (into #{} (map second)
```

Comment:
This comment is worth highlighting. `[:does-not-contain 'field "bob"]` returns true for all records with the field which don't match "bob" as well as records without 'field. This might be a bug in our mbql where we should emit that the field exists and it does not contain "bob"?

Reply:
It's a little tricky because MongoDB doesn't have a formal schema the same way SQL databases do. I'm not really sure what we should do with filter operators like these for DBs with dynamic schemas if the key doesn't exist at all. I think we should open an issue and think about it there... I wouldn't change the existing behavior for this PR